avivator/src/utils.js

```diff
@@ -65,7 +65,7 @@ export async function createLoader(
       // Non-Bioformats6 pyramids use Image tags for pyramid levels and do not have offsets
       // built in to the format for them, hence the ternary.
       const {
-        omexml: { SizeZ, SizeT, SizeC },
+        metadata: { SizeZ, SizeT, SizeC },
```

Comment:
A `getDimensionSizes` method for the loaders seems appropriate, right?  Instead of the above, we could implement something that does this:
```javascript
const sizeZ = this.dimensions.find(dim => dim.field === 'z').values.length;
const sizeT = this.dimensions.find(dim => dim.field === 'time').values.length;
const sizeC = this.dimensions.find(dim => dim.field === 'channel').values.length;
```

Reply:
Hopefully I have time to review this over the week but in general I want to move away from `dimensions` on loaders. 

This field isn't used by any of Viv's layers and instead only for the UI to create `loaderSelections`. I'd rather have an `openOMETiff(url)` function that returns: `{ loader, metadata }`. Rather than encapsulating the metadata within the loader object, not all pixel sources have labeled axes.

Reply:
Basically I just want the loader to describe the nd-source, but not contain additional metadata. `getDimensionSizes` seems like an appropriate method, but requiring all dimensions to be labeled feels like an error in our design. I'd rather the loader just have a `shape` property which describes the shape of the nd-array source. Then, the metadata could describe what those dimensions are:

```javascript
const metadata = {
  axis_labels: ['time', 'channel', 'z', 'y', 'x'],
  ...other_metadata,
}
```

Or we could simplify the dimensions the drastically:

```javascript
const dimensions = [
  'time',
  { name: 'channel', values: [/*...*/] },  // if there are values, it's an object, otherwise just a string
  'z',
  'y',
  'x'
];
```

I think the original spec is overly complicated with `type`. What is it used for? Additionally, we just use `range(SizeT)` to create values that are only used in the UI. Maybe it isn't possible for OME-TIFF..

Reply:
I'm fine with this but I have two questions/comments:
1. How will `serializeSelection` and `getImages` work without any notion of the labels for dimensions?  Will the loader still have this?
2. Can we merge this independent of that since I am probably not the right person to implement this (you seem to have a very strong concept in mind) and this PR has value on its own in the interim since it both aligns our loaders to some degree and resolves a separate outstanding issue?

Reply:
1.) I probably need some more time to think on it... starting to think labeled dimensions is a reasonable, but still we could simplify the current `dimensions` object a lot. Moving towards just labels for dimensions on the loaders (no `type` and maybe optional `values`) would be a good step in my opinion. Perhaps take some inspiration from [tensorstore](https://google.github.io/tensorstore/python/indexing.html#dimension-expressions).

2.) Yes. I'll review tomorrow in the AM. This doesn't change much of the interfaces that already exist on the loaders, I just want to voice some hesitance towards adding more loader methods.